lua/nvim-treesitter/query.lua

```diff
@@ -46,8 +46,56 @@ function M.has_query_files(lang, query_name)
   return files and #files > 0
 end
 
-function M.get_query(lang, query_name)
-  return tsq.get_query(lang, query_name)
+do
+  local mt = {}
+  mt.__index = function(tbl, key)
+    if rawget(tbl, key) == nil then
+      rawset(tbl, key, {})
+    end
+    return rawget(tbl, key)
+  end
```

Comment:
Don't we have utilities for caching in `caching.lua` ?

Reply:
That seems to be for caching on each buffer while this is for a global cache of the queries files.  For example the function that uses `caching.lua` is`get_matches`. That will run `update_cached_matches` which runs `iter_group_results` which will get the query files for the current buffer using `get_query` which is the function that I am implementing caching for.